src/hooks/usePokemons.ts

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { PokemonType } from '@/types/pokemonType';
+import { fetchAllPokemons } from '@/api/pokemonApi';
+
+interface UsePokemonsAPI {
+  pokemons: PokemonType[];
+  errorArr: Error[];
+  ignoreFetch: boolean;
+}
+
+export const usePokemons = (): UsePokemonsAPI => {
+  const [pokemons, setPokemons] = useState<PokemonType[]>([]);
+  const [errorArr, setErrArr] = useState<Error[]>([]);
```

Comment:
```suggestion
  const [errors, setErrors] = useState<Error[]>([]);
```

We're in **Type**script after all. The type is there.